### PR TITLE
Doc update of Python 3.6 to 3.7

### LINKED ISSUE #6549: Syntax error on python 3.6 and on 3.8
**CKAN version**
master

**Describe the bug**
Running cli commands results in syntax error with python 3.6
```
SyntaxError: future feature annotations is not defined
```

We could either fix the error or bump our minimum python version to 3.7 as 3.6 end of life date is in a month anyway. 

On python 3.8, master produces different error:

```
from typing_extensions import TypedDict
ModuleNotFoundError: No module named 'typing_extensions'
```

### PR DESCRIPTION
**Pull request by Gauravp-NEC**
Fixes #6549 

### Features:

- [ ] includes tests covering changes
- [x] includes updated documentation
- [ ] includes user-visible changes
- [ ] includes API changes
- [ ] includes bugfix for possible backport

Please [X] all the boxes above that apply

### CODE REVIEW

**Comment by Gauravp-NEC** [comment]
@kowh-ai , Please review this PR. Thanks

**Comment by Gauravp-NEC** [comment]
@kowh-ai , Please review this PR. Thanks

**Comment by Gauravp-NEC** [comment]
@kowh-ai , Please review this PR as it is open for so long. Thanks :)

**Comment by Gauravp-NEC** [comment]
@kowh-ai , Please review this PR, it is open since many days.

File: doc/maintaining/installing/index.rst [hunk 1]
```diff
@@ -9,7 +9,7 @@ There are three ways to install CKAN:
 #. Install from source
 #. Install from Docker Compose
 
-CKAN 2.9 supports Python 3.6 or higher and Python 2.7. The next version of CKAN
+CKAN 2.9 supports Python 3.7 or higher and Python 3.8. The next version of CKAN
```
Comment (kowh-ai) [inline doc/maintaining/installing/index.rst:12]: Should be CKAN 2.9 supports Python 3.7 and 3.8 _plus_ 2.7

File: doc/maintaining/installing/install-from-package.rst [hunk 2]
```diff
@@ -5,18 +5,18 @@ Installing CKAN from package
 ============================
 
 This section describes how to install CKAN from package. This is the quickest
-and easiest way to install CKAN, but it requires **Ubuntu 18.04 (Python 2) or 20.04 (Python 3 or Python 2) 64-bit**. If
+and easiest way to install CKAN, but it requires **Ubuntu 18.04 (Python 3.7) or 20.04 (Python 3.8) 64-bit**. If
```
Comment (kowh-ai) [inline doc/maintaining/installing/install-from-package.rst:8]: Best not to mention Python versions here as Ubuntu 18.04 comes with Python 3.6 (not 3.7)

File: doc/maintaining/installing/install-from-package.rst [hunk 3]
```diff
@@ -5,18 +5,18 @@ Installing CKAN from package
...
 At the end of the installation process you will end up with two running web
 applications, CKAN itself and the DataPusher, a separate service for automatically
 importing data to CKAN's :doc:`/maintaining/datastore`. Additionally, there will be a process running the worker for running :doc:`/maintaining/background-tasks`. All these processes will be managed by `Supervisor <https://supervisord.org/>`_.
 
-For Python 3 installations, the minimum Python version required is 3.6.
+For Python 3 installations, the minimum Python version required is 3.7.
 
 * **Ubuntu 20.04** includes **Python 3.8** as part of its distribution
-* **Ubuntu 18.04** includes **Python 3.6** as part of its distribution
+* **Ubuntu 18.04** includes **Python 3.7** as part of its distribution
```
Comment (kowh-ai) [inline doc/maintaining/installing/install-from-package.rst:19]: Nope, comes with Python 3.6

**Review by kowh-ai — APPROVED** [review]
Looks fine - thanks!

**Pull request merged**